lib/taskana-core/src/main/java/pro/taskana/task/internal/TaskCommentServiceImpl.java

```diff
@@ -0,0 +1,268 @@
+
+    LOGGER.debug("entry to getTaskComments (taskId = {})", taskId);
+
+    try {
+
+      taskanaEngine.openConnection();
+
+      taskService.getTask(taskId);
+
+      List<TaskComment> taskComments =
+          taskCommentMapper.findByTaskId(taskId).stream().collect(Collectors.toList());
+
+      if (taskComments.isEmpty()) {
+        LOGGER.debug("getTaskComments() found no comments for the provided taskId");
+      }
+
+      return taskComments;
+
+    } finally {
+
+      taskanaEngine.returnConnection();
+
+      LOGGER.debug("exit from getTaskComments()");
+    }
+  }
+
+  TaskComment getTaskComment(String taskCommentId)
+      throws TaskCommentNotFoundException, NotAuthorizedException, TaskNotFoundException {
+
+    LOGGER.debug("entry to getTaskComment (taskCommentId = {})", taskCommentId);
+
+    TaskCommentImpl result = null;
+
+    try {
+
+      taskanaEngine.openConnection();
+
+      result = taskCommentMapper.findById(taskCommentId);
+
+      if (result == null) {
+        throw new TaskCommentNotFoundException(
+            taskCommentId, "TaskComment for id " + taskCommentId + " was not found");
+      }
+
+      taskService.getTask(result.getTaskId());
+
+      return result;
+
+    } finally {
+
+      taskanaEngine.returnConnection();
+
+      LOGGER.debug("exit from getTaskComment()");
+    }
+  }
+
+  private void checkModifiedHasNotChanged(
+      TaskComment oldTaskComment, TaskComment taskCommentImplToUpdate) throws ConcurrencyException {
+
+    if (!oldTaskComment.getModified().equals(taskCommentImplToUpdate.getModified())) {
+
+      throw new ConcurrencyException(
+          "The current TaskComment has been modified while editing. "
+              + "The values can not be updated. TaskComment "
+              + taskCommentImplToUpdate.toString());
+    }
+  }
+
+  private void initDefaultTaskCommentValues(TaskCommentImpl taskCommentImplToCreate) {
+
+    Instant now = Instant.now();
+
+    taskCommentImplToCreate.setId(IdGenerator.generateWithPrefix(ID_PREFIX_TASK_COMMENT));
+    taskCommentImplToCreate.setModified(now);
+    taskCommentImplToCreate.setCreated(now);
+
+    String creator = CurrentUserContext.getUserid();
+    if (taskanaEngine.getEngine().getConfiguration().isSecurityEnabled() && creator == null) {
+      throw new SystemException(
+          "TaskanaSecurity is enabled, but the current UserId is"
+              + " NULL while creating a TaskComment.");
+    }
+    taskCommentImplToCreate.setCreator(creator);
+  }
+
+  private void validateNoneExistingTaskCommentId(String taskCommentId)
+      throws InvalidArgumentException {
+
+    if (taskCommentId != null && !taskCommentId.equals("")) {
+      throw new InvalidArgumentException(
+          String.format(
+              "taskCommentId must be null/empty for creation, but found %s", taskCommentId));
+    }
+  }
+
+  private void validateExistingTaskCommentId(String taskCommentId) throws InvalidArgumentException {
```

Comment:
I don't like the method name 'validateExistingTaskCommentId' because it implies that it validates that the id 'exists'. However, it simply validates that the id is not null or empty. I would propose to find a name that better describes what the method does, e.g. verifyTaskCommentIdIsNotEmpty or something similar

Reply:
changed the name to verifyTaskCommentIdIsNotNullOrEmpty